impr: added character-counter component to show how many characters are left in bio and keyboard fields on profile-edit page (theiereman)

theiereman opened the pull request:
### Description

I noticed a backend error when submitting a keyboard description that was too long (75 max), so I thought it would be nice to prevent backend errors and provide some kind of feedback about the number of characters left before submitting a value.

![firefox_KG2Mu1W79i](https://github.com/monkeytypegame/monkeytype/assets/141121240/17308bb2-8416-4a50-a764-c10c55bc8f29)

Miodec commented:
We already have a character counter on the report modal - can you make it look the same? 
![image](https://github.com/monkeytypegame/monkeytype/assets/13181393/1829da45-9431-4688-8d36-ff81de87d558)

You could update it to `x/(character limit)` too on the report and profile update screens

Miodec commented on frontend/src/ts/elements/character-counter.ts:
```diff
@@ -0,0 +1,47 @@
+    this.counterElement.text(`${remaining.toString()} characters remaining`);
+
+    const remainingPercentage = (remaining / this.maxLength) * 100;
+    this.counterElement.css("color", remainingPercentage < 10 ? "red" : "");
```
This should use theme error color instead of red.

Miodec commented:
Alternatively you could update the report modal to use your new system.

theiereman commented:
I changed every character counter I found (user report ; quote report ; quote submit) to the new system and adjusted the style of the component to fit the old one (except the "x/max length" part).

Miodec reviewed (approved):
Thanks
